03-task-manager/starter/controllers/tasks.js

```diff
@@ -25,7 +25,7 @@ exports.getTask = asyncWrapper(async (req, res, next) => {
 
 exports.updateTask = asyncWrapper(async (req, res) => {
   const { id } = req.params;
-  const task = await Task.findByIdAndUpdate(id, req.body, {
+  const task = await Task.findOneAndUpdate(id, req.body, {
```

Comment:
FYI: what you had before was good!

`findByIdAndUpdate` and `findByIdAndDelete`, take 3 arguments:
1. the id that you want to use to look up the object in Mongo
2. the object representing the updates you want to make
3. additional options

Now that you've switched to `findOneAndUpdate` and `findOneAndRemove`, the first argument is slightly different!:
1. a query object representing what fields you want to look up by. So instead of just passing `id` by itself. You'd now need to pass in `{ _id: id }` to tell Mongo to look for documents whose `_id` field matches the `id` variable you're passing in
2. the object representing the updates you want to make
3. additional options

I haven't tried your current code out myself yet but I believe it wouldn't work with the changes you're made here

Reply:
Actually, it does work, but I will change it since it is not the best practice.

Reply:
Ah yes, looks like you're correct; never mind me :sweat\_smile: !